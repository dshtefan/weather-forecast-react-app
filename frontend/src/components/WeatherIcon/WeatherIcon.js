import React from 'react';
import icon01 from './icons/01.svg';
import icon02 from './icons/02.svg';
import icon03 from './icons/03.svg';
import icon04 from './icons/04.svg';
import icon09 from './icons/09.svg';
import icon10 from './icons/10.svg';
import icon11 from './icons/11.svg';
import icon13 from './icons/13.svg';
import icon50 from './icons/50.svg';

const WeatherIcon = ({iconNumber}) => {
  const createImg = (icon) => (
    <img src={icon} alt=""/>
  );
  switch (iconNumber) {
    case '01':
      return createImg(icon01);
    case '02':
      return createImg(icon02);
    case '03':
      return createImg(icon03);
    case '04':
      return createImg(icon04);
    case '09':
      return createImg(icon09);
    case '10':
      return createImg(icon10);
    case '11':
      return createImg(icon11);
    case '13':
      return createImg(icon13);
    case '50':
      return createImg(icon50);
    default :
      return null;
  }
};

export default WeatherIcon;