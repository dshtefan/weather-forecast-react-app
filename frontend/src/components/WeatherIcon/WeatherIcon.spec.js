import React from 'react';
import WeatherIcon from "./WeatherIcon";

describe('WeatherIcon', () => {
  const numbers = ['01', '02', '03', '04', '09', '10', '11', '13', '50'];
  const iconNumber = undefined;

  describe('дефолтный кейс', () => {
    const weatherIcon = shallow(<WeatherIcon iconNumber={iconNumber} />);
    it ('отображается корректно', () => {
      expect(shallowToJson(weatherIcon)).toMatchSnapshot();
    });
  });

  describe('иконки', () => {
    for (let i = 0; i < numbers.length; i++) {
      const weatherIcon = shallow(<WeatherIcon iconNumber={numbers[i]} />);
      it ('отображаются корректно', () => {
        expect(shallowToJson(weatherIcon)).toMatchSnapshot();
      });
    }
  });
});