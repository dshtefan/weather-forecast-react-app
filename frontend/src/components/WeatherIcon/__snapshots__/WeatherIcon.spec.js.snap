// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityListItem дефолтный кейс отображается корректно 1`] = `""`;

exports[`CityListItem иконки отображаются корректно 1`] = `
<img
  alt=""
  src="01.svg"
/>
`;

exports[`CityListItem иконки отображаются корректно 2`] = `
<img
  alt=""
  src="02.svg"
/>
`;

exports[`CityListItem иконки отображаются корректно 3`] = `
<img
  alt=""
  src="03.svg"
/>
`;

exports[`CityListItem иконки отображаются корректно 4`] = `
<img
  alt=""
  src="04.svg"
/>
`;

exports[`CityListItem иконки отображаются корректно 5`] = `
<img
  alt=""
  src="09.svg"
/>
`;

exports[`CityListItem иконки отображаются корректно 6`] = `
<img
  alt=""
  src="10.svg"
/>
`;

exports[`CityListItem иконки отображаются корректно 7`] = `
<img
  alt=""
  src="11.svg"
/>
`;

exports[`CityListItem иконки отображаются корректно 8`] = `
<img
  alt=""
  src="13.svg"
/>
`;

exports[`CityListItem иконки отображаются корректно 9`] = `
<img
  alt=""
  src="50.svg"
/>
`;

exports[`WeatherIcon дефолтный кейс отображается корректно 1`] = `""`;

exports[`WeatherIcon иконки отображаются корректно 1`] = `
<img
  alt=""
  src="01.svg"
/>
`;

exports[`WeatherIcon иконки отображаются корректно 2`] = `
<img
  alt=""
  src="02.svg"
/>
`;

exports[`WeatherIcon иконки отображаются корректно 3`] = `
<img
  alt=""
  src="03.svg"
/>
`;

exports[`WeatherIcon иконки отображаются корректно 4`] = `
<img
  alt=""
  src="04.svg"
/>
`;

exports[`WeatherIcon иконки отображаются корректно 5`] = `
<img
  alt=""
  src="09.svg"
/>
`;

exports[`WeatherIcon иконки отображаются корректно 6`] = `
<img
  alt=""
  src="10.svg"
/>
`;

exports[`WeatherIcon иконки отображаются корректно 7`] = `
<img
  alt=""
  src="11.svg"
/>
`;

exports[`WeatherIcon иконки отображаются корректно 8`] = `
<img
  alt=""
  src="13.svg"
/>
`;

exports[`WeatherIcon иконки отображаются корректно 9`] = `
<img
  alt=""
  src="50.svg"
/>
`;
